### PR TITLE
[Graphbolt] Add to dgl

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
To dgl used to converting a minibatch to a dgl minbatch used for computing.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/minibatch.py [hunk 1]
```diff
@@ -290,27 +275,106 @@ def to_dgl_blocks(self):
...
                     block.edata[dgl.EID] = subgraph.original_edge_ids
-
         return blocks
 
-    def __repr__(self) -> str:
-        return _minibatch_str(self)
+    def to_dgl(self):
+        """ "Converting a `MiniBatch` into a DGL MiniBatch that contains
+        everything necessary for computation."
+        """
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:290]:
```suggestion
        """Converting a `MiniBatch` into a DGL MiniBatch that contains
        everything necessary for computation.
        """
```

File: python/dgl/graphbolt/minibatch.py [hunk 2]
```diff
@@ -290,27 +275,106 @@ def to_dgl_blocks(self):
...
+            blocks=self._to_dgl_blocks(),
+            input_nodes=self.input_nodes,
+            output_nodes=self.seed_nodes,
+            node_features=self.node_features,
+            edge_features=self.edge_features,
+            labels=self.labels,
+        )
+        assert (
+            minibatch.blocks is not None
+        ), "Sampled subgraphs for computation are Missing."
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:301]:
```suggestion
        ), "Sampled subgraphs for computation are missing."
```

File: python/dgl/graphbolt/minibatch.py [hunk 3]
```diff
@@ -290,27 +275,106 @@ def to_dgl_blocks(self):
...
+
+        # For link prediction tasks.
+        if self.compacted_node_pairs is not None:
+            assert (
+                self.compacted_node_pairs is not None
+            ), "Evaluate graph for Computation in link prediction task is Missing."
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:307]: if and then assert? Does this assert actually required?

File: tests/python/pytorch/graphbolt/impl/test_minibatch.py [hunk 4]
```diff
@@ -78,176 +135,133 @@ def test_to_dgl_blocks_hetero():
...
     for i, block in enumerate(blocks):
         assert torch.equal(block.edges()[0], node_pairs[i][0])
         assert torch.equal(block.edges()[1], node_pairs[i][1])
         assert torch.equal(block.edata[dgl.EID], original_edge_ids[i])
-        assert torch.equal(block.edata["x"], edge_features[i]["x"])
     assert torch.equal(blocks[0].srcdata[dgl.NID], original_row_node_ids[0])
-    assert torch.equal(blocks[0].srcdata["x"], node_features["x"])
 
 
-def test_representation():
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_minibatch.py:142]: Add this test case back.

File: tests/python/pytorch/graphbolt/impl/test_minibatch.py [hunk 5]
```diff
@@ -78,176 +135,133 @@ def test_to_dgl_blocks_hetero():
...
+    # Assert
+    assert len(dgl_minibatch.blocks) == 2
+    assert minibatch.node_features is dgl_minibatch.node_features
+    assert minibatch.edge_features is dgl_minibatch.edge_features
+    assert minibatch.labels is dgl_minibatch.labels
+    assert minibatch.seed_nodes is dgl_minibatch.output_nodes
+    check_dgl_blocks_homo(minibatch, dgl_minibatch.blocks)
+
+
+def test_to_node_hetero():
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_minibatch.py:175]:
```suggestion
def test_to_dgl_node_classification_hetero():
```

File: tests/python/pytorch/graphbolt/impl/test_minibatch.py [hunk 6]
```diff
@@ -78,176 +135,133 @@ def test_to_dgl_blocks_hetero():
...
+    assert len(dgl_minibatch.blocks) == 2
+    assert minibatch.node_features is dgl_minibatch.node_features
+    assert minibatch.edge_features is dgl_minibatch.edge_features
+    assert minibatch.labels is dgl_minibatch.labels
+    assert minibatch.seed_nodes is dgl_minibatch.output_nodes
+    check_dgl_blocks_hetero(minibatch, dgl_minibatch.blocks)
+
+
+@pytest.mark.parametrize("mode", ["neg_graph", "neg_src", "neg_dst"])
+def test_to_dgl_link_homo(mode):
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_minibatch.py:191]:
```suggestion
def test_to_dgl_link_predication_homo(mode):
```

File: tests/python/pytorch/graphbolt/impl/test_minibatch.py [hunk 7]
```diff
@@ -78,176 +135,133 @@ def test_to_dgl_blocks_hetero():
...
+        )
+    if mode == "neg_graph" or mode == "neg_dst":
+        assert torch.equal(
+            dgl_minibatch.negative_node_pairs[1],
+            minibatch.compacted_negative_dsts.view(-1),
+        )
+
+
+@pytest.mark.parametrize("mode", ["neg_graph", "neg_src", "neg_dst"])
+def test_to_dgl_link_hetero(mode):
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_minibatch.py:224]:
```suggestion
def test_to_dgl_link_predication_hetero(mode):
```

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/minibatch.py (hunk 3 above)
Comment (peizhou001) [reply python/dgl/graphbolt/minibatch.py:307]: No，should be removed

**Pull request merged**